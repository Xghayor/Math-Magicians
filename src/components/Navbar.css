/* Navbar container */
.navbar {
  background-color: #333;
  color: white;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 1rem;
}

/* Navbar brand/logo */
.navbar-brand a {
  color: white;
  text-decoration: none;
  font-size: 1.5rem;
  font-weight: bold;
}

/* Navbar menu items */
.navbar-menu {
  list-style: none;
  display: flex;
  gap: 1rem;
}

.navbar-item a {
  cursor: pointer;
  color: white;
  text-decoration: none;
  transition: color 0.3s ease-in-out;
}

.navbar-item a:hover {
  color: #f90;
}

@media (max-width: 768px) {
  .navbar {
    flex-direction: column;
    align-items: flex-start;
  }

  .navbar-menu {
    margin-top: 1rem;
  }

  .navbar-item {
    display: block;
    padding: 0.5rem 0;
  }
}
