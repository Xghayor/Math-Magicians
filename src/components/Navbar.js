import React from 'react';
import { Link } from 'react-router-dom';
import './Navbar.css';

const Navbar = () => (
  <nav className="navbar">
    <div className="navbar-brand">
      <Link to="/">Math Magicians</Link>
    </div>
    <ul className="navbar-menu">
      <li className="navbar-item">
        <Link to="/">Home</Link>
      </li>
      <li className="navbar-item">
        <Link to="/Calculator">Calculator</Link>
      </li>
      <li className="navbar-item">
        <Link to="/Quote">Quote</Link>
      </li>
    </ul>
  </nav>
);

export default Navbar;
