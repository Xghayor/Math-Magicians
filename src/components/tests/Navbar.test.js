import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Navbar from '../Navbar';

test('Navbar component matches snapshot', () => {
  const { asFragment } = render(
    <MemoryRouter>
      <Navbar />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
