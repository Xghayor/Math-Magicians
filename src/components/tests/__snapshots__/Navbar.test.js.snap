// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar"
  >
    <div
      class="navbar-brand"
    >
      <a
        href="/"
      >
        Math Magicians
      </a>
    </div>
    <ul
      class="navbar-menu"
    >
      <li
        class="navbar-item"
      >
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="navbar-item"
      >
        <a
          href="/Calculator"
        >
          Calculator
        </a>
      </li>
      <li
        class="navbar-item"
      >
        <a
          href="/Quote"
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
